Maple2.Server.Game/Manager/HousingManager.cs

```diff
@@ -351,6 +352,145 @@ public void InitNewHome(string characterName, ExportedUgcMapMetadata? template)
         db.SaveCubes(Home.Indoor, plotCubes);
     }
 
+    private readonly Dictionary<HousingCategory, int> decorationGoal = new() {
+        { HousingCategory.Bed, 1 },
+        { HousingCategory.Table, 1 },
+        { HousingCategory.SofasChairs, 2 },
+        { HousingCategory.Storage, 1 },
+        { HousingCategory.WallDecoration, 1 },
+        { HousingCategory.WallTiles, 3 },
+        { HousingCategory.Bathroom, 1 },
+        { HousingCategory.Lighting, 1 },
+        { HousingCategory.Electronics, 1 },
+        { HousingCategory.Fences, 2 },
+        { HousingCategory.NaturalTerrain, 4 },
+    };
+
+    public void InteriorCheckIn(Plot plot) {
+        if (session.Item.Inventory.FreeSlots(InventoryType.Misc) < 1 || session.Item.Inventory.FreeSlots(InventoryType.Fragment) < 1) {
+            return;
+        }
+
+        Dictionary<HousingCategory, int> decorationCurrent = plot.Cubes.Values.GroupBy(plotCube => plotCube.HousingCategory)
+            .ToDictionary(grouping => grouping.Key, grouping => grouping.Count());
+
+        int decorationScore = 0;
+
+        foreach (KeyValuePair<HousingCategory, int> goal in decorationGoal) {
+            if (!decorationCurrent.TryGetValue(goal.Key, out int current)) {
+                continue;
+            }
+
+            if (current < goal.Value) {
+                continue;
+            }
+
+            switch (goal.Key) {
+                case HousingCategory.Bed:
+                case HousingCategory.SofasChairs:
+                case HousingCategory.WallDecoration:
+                case HousingCategory.WallTiles:
+                case HousingCategory.Bathroom:
+                case HousingCategory.Lighting:
+                case HousingCategory.Fences:
+                case HousingCategory.NaturalTerrain:
+                    decorationScore += 100;
+                    break;
+                case HousingCategory.Table:
+                case HousingCategory.Storage:
+                    decorationScore += 50;
+                    break;
+                case HousingCategory.Electronics:
+                    decorationScore += 200;
+                    break;
+                default:
+                    continue;
+            }
+        }
+
+        int housingPoint = decorationScore switch {
+            < 300 => 100,
+            < 500 => 300,
+            < 700 => 500, // 500 and 700 are the same reward for some reason
+            < 1100 => 700,
+            _ => 1100,
+        };
+
+        tableMetadata.UgcHousingPointRewardTable.Entries.TryGetValue(housingPoint, out UgcHousingPointRewardTable.Entry? reward);
+        if (reward is null) {
+            logger.Error("Failed to get reward for decoration score {DecorationScore}.", housingPoint);
+            return;
+        }
+
+        serverTableMetadata.IndividualDropItemTable.Entries.TryGetValue(reward.IndividualDropBoxId, out IDictionary<int, IndividualDropItemTable.Entry>? entry);
```

Comment:
why aren't we using sesssion.Field.ItemDrop.GetIndividualDropboxItems ?

Reply:
when i tried it returned the list of items without the weights

Reply:
it does the weighing for you